Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+            The domain passed to the new Table object.
+
+        Returns
+        -------
+        TableBase
+            The new Table with a Domain assigned.
+        """
+        result = cls(columns=domain.attributes + domain.class_vars + domain.metas)
+        result.domain = domain
+        return result
+
+    @classmethod
+    @deprecated("t.iloc[row_indices].copy()")
+    def from_table_rows(cls, source, row_indices):
+        """Construct a new Table as a copy of the selected rows (position indexing).
+
+        Parameters
+        ----------
+        source : TableBase
+            The source Table.
+        row_indices
+            The locations of the selected rows in any of the standard formats.
+        Returns
+        -------
+        TableBase
+            A subset of the original Table as a copy.
+        """
+        # don't just plain copy here: in case of subclasses of Table, a plain table is passed
+        # through the constructor and from_table to here, and expects to be converted
+        # into a proper subclass type
+        result = cls(data=source.iloc[row_indices]).copy()
+        result._transfer_properties(source, transfer_domain=True)  # because we manually copy data, not the whole table
+        return result
+
+    @classmethod
+    def from_table(cls, target_domain, source_table, row_indices=slice(None)):
+        """Create a new Table as a conversion from the source Table's domain to the new one.
+
+        Row indices may be given to select a subset at the same time.
+        The new domain defines the columns of the new table - new columns are computed
+        on the fly from source columns whenever appropriate.
+
+        Parameters
+        ----------
+        target_domain : Domain
+            The new domain. Defines the resulting columns and the transformation
+            from the original Table to the new one.
+        source_table : TableBase
+            The source table.
+        row_indices : slice | list[int] | list[bool], optional
+            The row indices to select a subset of the resulting table.
+
+        Returns
+        -------
+        TableBase
+            A table converted from the source domain to the target domain.
+        """
+        # if a series is passed, convert it to a frame
+        if isinstance(source_table, SeriesBase):
+            d = source_table.domain
+            # to pandas df first to avoid adding a column/row of weights automatically
+            source_table = cls(source_table.domain, pd.DataFrame(source_table).transpose())
+
+        new_cache = cls._conversion_cache is None
+        try:
+            if new_cache:
+                cls._conversion_cache = {}
+            else:
+                cached = cls._conversion_cache.get((id(target_domain), id(source_table)))
+                if cached is not None:
+                    return cached
+            if target_domain == source_table.domain:
+                # intentional casting to subclass
```

Comment:
I'll explain all three comments here. This is identical to the now-defunct `from_table_rows`. `cls` is used so we cast to a subclass (just passing through the data). A copy is not needed (I removed it). We then need to transfer the properties because when passing a table as `data`, `_metadata` things are not implicitly transferred. 

Any idea of how to cast to a subclass in a nicer way? The contract in this case is that is that `Corpus(existing_table)` must return a copy (which I don't like if we just want to cast) of type `Corpus` (not `type(existing_table)`).